core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -160,23 +148,96 @@ public InodeTree(ContainerIdGenerable containerIdGenerator,
    */
   public void initializeRoot(String owner, String group, Mode mode, JournalContext context)
       throws UnavailableException {
-    if (mRoot == null) {
+    if (mState.getRoot() == null) {
       InodeDirectory root = InodeDirectory.create(mDirectoryIdGenerator.getNewDirectoryId(context),
           NO_PARENT, ROOT_INODE_NAME,
           CreateDirectoryOptions.defaults().setOwner(owner).setGroup(group).setMode(mode));
-      setRoot(root);
+      root.setPersistenceState(PersistenceState.PERSISTED);
+      mState.applyAndJournal(context, root);
     }
   }
 
+  /**
+   * Applies a journal entry to the inode tree state. This method should only be used during journal
+   * replay.
+   *
+   * @param entry an entry to apply to the inode tree
+   */
+  public void replayJournalEntryFromJournal(JournalEntry entry) {
+    mState.replayJournalEntryFromJournal(entry);
+  }
+
+  /**
+   * @return the list of TTL buckets for tracking inode TTLs
+   */
+  public TtlBucketList getTtlBuckets() {
+    return mTtlBuckets;
+  }
+
+  /**
+   * Marks an inode directory as having its direct children loaded.
+   *
+   * @param context journal context supplier
+   * @param id the inode directory id
+   */
+  public void setDirectChildrenLoaded(Supplier<JournalContext> context, long id) {
```

Comment:
Is this just a shortcut for calling something like `updateInode`? (or the directory equivalent)

Should this take an InodeDirectoryView?

Reply:
good idea, done